Source/buildimplementationjs.go

```diff
@@ -767,7 +767,7 @@ func v8TypeString(paramType string) string {
 func generateReturnValue(param ComponentDefinitionParam, subComponent ComponentDefinition) string {
 	returnValue := ""
 	if param.ParamType == "class" {
-		returnValue = fmt.Sprintf("createV8Instance(objectCreator, param%s, %s_MapClassIdToInjectionClassType(param%s->%s()))", param.ParamName, subComponent.NameSpace, param.ParamName, subComponent.Global.ClassTypeIdMethod)
+		returnValue = fmt.Sprintf("param%s?createV8Instance(objectCreator, param%s, %s_MapClassIdToInjectionClassType(param%s->%s())):v8::Local<v8::Object>()", param.ParamName, param.ParamName, subComponent.NameSpace, param.ParamName, subComponent.Global.ClassTypeIdMethod) // Very bad
```

Comment:
yes, this is a huge mouth full and hard to read. but you are not introducing that pattern here.
I would even go as far as to remove the comment, or atleast say in the comment that you are not happy with the long ternary statement itself.
Would need some refactoring to get rid of...

Reply:
I think I left that in accidentally because I did something 'very bad', changed it and then left the comment.  I'll get it removed

Reply:
No its that line.  It is just hard to read.  I'll do something about that.